Userland/Libraries/LibWeb/Layout/BlockFormattingContext.cpp

```diff
@@ -124,8 +124,13 @@ void BlockFormattingContext::compute_width(Box const& box, AvailableSpace const&
         return;
     }
 
+    auto box_should_avoid_floats_because_it_establishes_fc = false;
+    if (auto formatting_context_type = formatting_context_type_created_by_box(box); formatting_context_type.has_value())
+        box_should_avoid_floats_because_it_establishes_fc = formatting_context_type.value() == Type::Block || formatting_context_type.value() == Type::Flex || formatting_context_type.value() == Type::Grid;
```

Comment:
Maybe this would look slightly nicer factored as a separate helper function.

```cpp
static bool box_should_avoid_floats_because_it_establishes_fc(Box const& box)
{
    ...
}
```

Reply:
yes, it looks nicer with helper :)